File: main/app/worker/get_user_by_id.py
import os
import boto3
from dotenv import load_dotenv
from botocore.exceptions import ClientError
load_dotenv()

AWS_DYNAMO = {
    'region_name': os.getenv('REGION'),
    'aws_access_key_id': os.getenv('ACCESS_KEY'),
    'aws_secret_access_key': os.getenv('SECRET_ACCESS_KEY'),
}

dynamodb = boto3.resource('dynamodb', **AWS_DYNAMO)


def get_user_by_id(user_id):
    try:
        table = dynamodb.Table(f"users-{os.getenv('STAGE')}")
        items = []
        last_evaluated_key = None
        while True:
            if last_evaluated_key:
                response = table.scan(ExclusiveStartKey=last_evaluated_key)
            else:
                response = table.scan()
            items.extend(response["Items"])
            last_evaluated_key = response.get("LastEvaluatedKey")
            if not last_evaluated_key:
                break
        for item in items:
            if item["user_id"] == user_id:
                return item
        return None
    except ClientError as err:
        print(f"Error getting user by id: {err}")
        return None

File: main/app/recognizer audb/common/getUserById.mjs
// getUserById.mjs 

import { DynamoDBDocument } from "@aws-sdk/lib-dynamodb";
import { DynamoDB } from "@aws-sdk/client-dynamodb";

const getUserById = async (user_id) => {
    try {
        const documentClient = DynamoDBDocument.from(new DynamoDB({ region: process.env.REGION }));
        const params = {
            TableName: `regroovio-users-${process.env.STAGE}`,
            Key: {
                user_id: user_id,
            },
        };
        const users = await documentClient.scan(params);
        if (users.Items.length === 0) {
            throw new Error(`No user with id ${user_id} found`);
        }
        return users.Items.find(item => item.user_id === user_id);
    } catch (err) {
        console.log(`Error getUserById: ${err}`);
        throw err;
    }
};
export { getUserById };
